pkg/controller/storagecluster/external_resources.go

```diff
@@ -141,9 +205,17 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterResources(scs []*s
 				// Setting the PoolName for RBD StorageClass
 				sc = scs[1]
 			} else if d.Name == cephRgwStorageClassName {
+				rgwEndpoint := d.Data[externalCephRgwEndpointKey]
+				extCephObjectStores, err = r.newExternalCephObjectStoreInstances(instance, rgwEndpoint, reqLogger)
+				if err != nil {
+					return err
+				}
+				// rgw-endpoint is no longer needed in the 'd.Data' dictionary,
+				// and can be deleted
+				delete(d.Data, externalCephRgwEndpointKey)
```

Comment:
What's the point of doing this? What happens if you don't delete it?

Reply:
RGW endpoint details we already stored in a `CephObjectStore`. If you don't `delete()` , the rgwEndpoint key / value will be again added to the StorageClass' `Parameters` map, which is not needed.

In the previous design, we used to store the rgw-endpoint details in the StorageClass itself, but now rgw-endpoint details are stored in a `CephObjectStore` object and in the SC, as one of the parameter, we just point to this cephobjectstore's name.

Reply:
...so then why is this still being passed in the RGW StorageClass data? There should be a separate place for it so that this hack isn't necessary.

Reply:
What I understood was,
We didn't make any change in the rook's JSON external input creation script (and kept it backward compatible). Thus changes were done only on OCS-Operator side with the same receiving input file.

Reply:
Thank you!!